app/controllers/AuthenticationController.scala

```diff
@@ -370,45 +372,55 @@ class AuthenticationController @Inject()(
   }
 
   def singleSignOn(sso: String, sig: String): Action[AnyContent] = sil.UserAwareAction.async { implicit request =>
-    if (ssoKey == "")
-      logger.warn("No SSO key configured! To use single-sign-on a sso key needs to be defined in the configuration.")
-
-    request.identity match {
-      case Some(user) =>
-        // logged in
-        // Check if the request we received was signed using our private sso-key
-        if (MessageDigest.isEqual(shaHex(ssoKey, sso).getBytes(StandardCharsets.UTF_8),
-                                  sig.getBytes(StandardCharsets.UTF_8))) {
-          val payload = new String(Base64.decodeBase64(sso))
-          val values = play.core.parsers.FormUrlEncodedParser.parse(payload)
-          for {
-            nonce <- values.get("nonce").flatMap(_.headOption) ?~> "Nonce is missing"
-            returnUrl <- values.get("return_sso_url").flatMap(_.headOption) ?~> "Return url is missing"
-            userEmail <- userService.emailFor(user)
-            _ = logger.info(f"User ${user._id} logged in via SSO.")
-          } yield {
-            val returnPayload =
-              s"nonce=$nonce&" +
-                s"email=${URLEncoder.encode(userEmail, "UTF-8")}&" +
-                s"external_id=${URLEncoder.encode(user._id.toString, "UTF-8")}&" +
-                s"username=${URLEncoder.encode(user.abbreviatedName, "UTF-8")}&" +
-                s"name=${URLEncoder.encode(user.name, "UTF-8")}"
-            val encodedReturnPayload = Base64.encodeBase64String(returnPayload.getBytes("UTF-8"))
-            val returnSignature = shaHex(ssoKey, encodedReturnPayload)
-            val query = "sso=" + URLEncoder.encode(encodedReturnPayload, "UTF-8") + "&sig=" + returnSignature
-            Redirect(returnUrl + "?" + query)
+    if (!isOIDCEnabled) {
+      Fox.successful(BadRequest("SSO is not enabled"))
+    } else {
```

Comment:
This should not be guarded by isOIDCEnabled, which is a feature flag for using a OIDC service to authenticate to WK (it correclty guards the other routes below).